modules/apigee_edge_teams/apigee_edge_teams.install

```diff
@@ -41,18 +41,36 @@ function apigee_edge_teams_requirements($phase) {
       $org_controller = \Drupal::service('apigee_edge.controller.organization');
       /* @var \Apigee\Edge\Api\Management\Entity\Organization $organization */
       $organization = $org_controller->load($sdk_connector->getOrganization());
-      if ($organization && !OrganizationFeatures::isCompaniesFeatureAvailable($organization) && OrganizationFeatures::isMonetizationEnabled($organization)) {
-        $url = [
-          ':url' => 'https://cloud.google.com/apigee/docs/api-platform/get-started/compare-apigee-products#unsupported-apis',
-        ];
-        $message = ($phase == 'runtime') ?
-          t("The Teams module functionality is not available for monetization enabled org on Apigee X / Hybrid and should be uninstalled, because <a href=':url' target='_blank'>AppGroup APIs are not supported in Apigee X / Hybrid orgs with monetization enabled</a>.", $url) :
-          t("The Teams module functionality is not available for monetization enabled org on Apigee X / Hybrid because <a href=':url' target='_blank'>AppGroup APIs are not supported in Apigee X / Hybrid orgs with monetization enabled</a>.", $url);
-        $requirements['apigee_edge_teams_not_supported'] = [
-          'title' => t('Apigee Edge Teams'),
-          'description' => $message,
-          'severity' => REQUIREMENT_ERROR,
-        ];
+      if ($organization && $org_controller->isOrganizationApigeeX()) {
+        // AppGroup APIs are not supported in Apigee X / Hybrid orgs with monetization enabled.
+        if ($organization->getAddonsConfig() || TRUE === $organization->getAddonsConfig()->getMonetizationConfig()->getEnabled()) {
+          $url = [
+            ':url' => 'https://cloud.google.com/apigee/docs/api-platform/get-started/compare-apigee-products#unsupported-apis',
+          ];
+          $message = ($phase == 'runtime') ?
+            t("The Teams module functionality is not available for monetization enabled org on Apigee X / Hybrid and should be uninstalled, because <a href=':url' target='_blank'>AppGroup APIs are not supported in Apigee X / Hybrid orgs with monetization enabled</a>.", $url) :
```

Comment:
I am really uncertain if this is actually the case as I wrote in the original issue. 

I have a monetization enabled trial Apigee X organization and the teams functionality is working without any issues. What should be missing is the Teams (AppGroups) and Monetization integration. This was the reason why I suggested to move this check to the Monetization Teams module. 

Can you please double check if there is any issue with the **base** Teams functionality on a Monetized Apigee X org?

Reply:
> I have a monetization enabled trial Apigee X organization and the teams functionality is working without any issues.

Ideally this shouldn't work as of now, because it is not completely supported in Apigee X (_but due the above code validation issue it got bypassed_). 
So basically for now we are blocking the teams module installing/enabling in Drupal with the rectified code `PR 940` and if in case, the Teams module is already enabled then a error message will be shown on all teams related pages (_ex: teams, team/{team_name}, teams settings etc_), rest will work as it is.
Added same code in M10n teams module as well.

@giteshk Can you please confirm.

Reply:
Thank you @shishir-intelli. That is how this module should behave.

Reply:
> Ideally this shouldn't work as of now, because it is not completely supported in Apigee X

So are we try to block enabling the Teams module on Apigee X because of these entries [from known limitations ](https://cloud.google.com/apigee/docs/api-platform/publish/organizing-client-app-ownership?hl=en#appgroups-limitations-and-known-issues)...?

> Metrics
> Quota enforcement
> Updates to the scope for API keys

I am just still trying to understand why this kill switch is here and not **only in apigee_m10n_teams** if the actual problem/limitations arises on a Monetization enabled organization (around Monetization features)? Why other customers on Apigee X should not use this feature now?

Reply:
Yes, We have to enforce the limitations.

Enabling Teams Module on monetized Apigee X orgs will set an expectation that Team Apps will follow the restrictions that Monetization Plans that users may have purchased.

We will enable this once AppGroup support for Monetization is announced.

Reply:
That basically means that customer who use the Teams feature on Apigee Edge cannot be migrated to Apigee X, they are blocked. 
This is a game changer in a conversation with a customer who uses team without Monetization and would like to move to Apigee X in this year...

Reply:
> That basically means that customers who use the Teams feature on Apigee Edge cannot be migrated to Apigee X, they are blocked.

No, They can migrate to Apigee X and use Teams module as well, only if Monetization is not enabled in the organisation which they are using.

eg:
```
        "monetizationConfig": {
            "enabled": false
        },
```

If 
```
        "monetizationConfig": {
            "enabled": true
        },
```
Monetization is enabled(TRUE) in the Org then the customer wont be able to use Teams module and Monetization Teams module in Apigee X.

Reply:
Ahh right, sorry, my bad, one morning coffee was not enough today...